concurrent-processing/concurrent_extraction.go

```diff
@@ -0,0 +1,142 @@
+/*
+ * This example demonstrates how to extract content form multiple documents concurrently
+ * with each document extraction running in its own go routine.
+ * N.B. currently concurrency is supported on a document level which means we can only extract one document per go routine.
+ *
+ * Run as: go run concurrent_extraction.go <input1.pdf> <input2.pdf> <input3.pdf>... <output_dir>
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/unidoc/unipdf/v3/common/license"
+	"github.com/unidoc/unipdf/v3/extractor"
+	"github.com/unidoc/unipdf/v3/model"
+)
+
+func init() {
+	// Make sure to load your metered License API key prior to using the library.
+	// If you need a key, you can sign up and create a free one at https://cloud.unidoc.io
+	err := license.SetMeteredKey(os.Getenv(`UNIDOC_LICENSE_API_KEY`))
+	if err != nil {
+		panic(err)
+	}
+}
+
+func main() {
+	if len(os.Args) < 3 {
+		fmt.Printf("Usage: go run concurrent_extraction.go input1.pdf input2.pdf input3.pdf... output_dir\n")
+		os.Exit(1)
+	}
+	inputDocuments := []string{}
+	args := os.Args
+	var outputDir string
+	for i := 1; i < len(args); i++ {
+		if i == len(args)-1 {
+			outputDir = args[i]
+		} else {
+			inputDocuments = append(inputDocuments, args[i])
+		}
+	}
+
+	start := time.Now()
+	runConcurrent(inputDocuments, outputDir)
+	duration := time.Since(start)
+	fmt.Println("time taken for concurrent extraction", duration)
+}
+
+// runConcurrent takes the list of input documents and destination output directory and runs the extraction concurrently.
+func runConcurrent(documents []string, outputDir string) {
+	res := make(chan map[string]string, 3)
```

Comment:
Shouldn't this same as document length? like
```
res := make(chan map[string]string, len(documents))
```

Reply:
Not necessarily. If we are reading as soon as the go routines are writting to the channel a buffer length of 1 is still enough but I made it 3 just to be safe. I think may be i should increase that in case reading from the channel is very slow due to the text writing to disk.

Reply:
@sampila Updated. Can you check again?